### ISSUE TITLE
Any query results in an error from the github API.

### ISSUE THREAD

**Issue by nathggns** [DefinitelyTyped/tsd#114]
Absolutely any query results in an error from the github API. This is a fatal bug and this software is currently unusable because of it.

```zsh
~ ❯ tsd query jquery

>> tsd 0.5.7
   missing required property: httpETag
      "httpETag" -> /required/3
       > object -> {"url":"https://api.github.com/repos/borisyankov/DefinitelyTyped/git/trees/51f5370b3e972cc065dba7...

-> an error occured!

Error: Missing required property: httpETag
    at new ValidationError (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:1133:12)
    at ValidatorContext.createError (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:156:9)
    at ValidatorContext.validateObjectRequiredProperties (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:721:22)
    at ValidatorContext.validateObject (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:688:11)
    at ValidatorContext.validateAll (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:394:11)
    at Object.api.validate (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:1233:24)
    at Object.api.validateResult (/usr/local/lib/node_modules/tsd/node_modules/tv4/tv4.js:1244:18)
    at JSONKoder.assert (/usr/local/lib/node_modules/tsd/build/api.js:4188:31)
    at /usr/local/lib/node_modules/tsd/build/api.js:4209:23
    at _fulfilled (/usr/local/lib/node_modules/tsd/node_modules/q/q.js:798:54)
```

**Comment by spazard1** [comment]
I'm running into the same problem.

**Comment by johnryan-wf** [comment]
same here

**Comment by spazard1** [comment]
VERY quick and dirty workaround in the meantime:
sudo vi /usr/lib/node_modules/tsd/build/api.js
:4206 // jumping to line 4206

Add this directly above the return Q().... line:
            value.httpETag = "*";

**Comment by evanweible-wf** [comment]
having this issue as well

**Comment by Bartvds** [comment]
Weird, did Github change anything? I can reproduce this on 0.5.7 but not on 0.6

So the easiest solution is to update to 0.6:

````
npm install tsd@next -g
````

Then run `tsd purge` to clear the cache, or if you wan to do it manually:

Run `tsd settings`: it will dump some data structure, look for the 'cacheDir' element, then navigate there and delete everything there (some json and a shitload of folders).

**Comment by Bartvds** [comment]
Follow-up: looks like Github stopped sending E-Tag http headers on the Github API.

I'll check Github support to see if it is a glitch or permanent, and verify if the date part of the http cache is picking up the slack properly.

**Comment by spazard1** [comment]
Annoyingly, gulp-tsd forces a version of 0.5.7...

**Comment by Bartvds** [comment]
Looks like the problem solved itself:

I got email back from Github support: they performed an infrastructure update today and had a problem with a dependency that stripped the E-Tag headers. It was fixed just now.

So I tested this locally, and I now see the headers and in TSD 0.5.7 things again works as expected.

If you try this with any version then you might need to purge the cache as described above.
